src/set_theory/ordinal/arithmetic.lean

```diff
@@ -2400,3 +2400,34 @@ meta def positivity_opow : expr → tactic strictness
 | _ := failed
 
 end tactic
+
+namespace acc
+variables {a b : α}
+
+/-- The length of the longest `r`-chain of elements ending in `a`. -/
+noncomputable def rank (h : acc r a) : ordinal :=
+acc.rec_on h $ λ a h ih, ordinal.sup $ λ b : {b // r b a}, order.succ $ ih b b.2
+
+lemma rank_eq (h : acc r a) :
+  h.rank = ordinal.sup (λ b : {b // r b a}, order.succ (h.inv b.2).rank) :=
+by { change (acc.intro a $ λ _, h.inv).rank = _, refl }
+
+/-- if `r a b` then the rank of `a` is less than the rank of `b`. -/
+lemma rank_lt_of_rel (hb : acc r b) (h : r a b) : (hb.inv h).rank < hb.rank :=
```

Comment:
Does it make sense to make a `@[strict_mono]` version of this (where `r = (<)`)?

Reply:
I added the `strict_mono`/`strict_anti` lemmas for `well_founded.rank`, but the `acc.rank` ones can't be stated because `acc.rank` is dependent.